client/settings/advanced-settings/index.js

```diff
@@ -0,0 +1,176 @@
+				label={ label }
+				checked={ isExpanded }
+				onChange={ toggleIsExpanded }
+			/>
+			{ isExpanded && <div ref={ wrapperRef }>{ children }</div> }
+		</>
+	);
+};
+
+const AdvancedSettings = () => {
+	const [ isSectionExpanded, toggleIsSectionExpanded ] = useToggle( false );
+	const firstHeadingElementRef = useRef( null );
+	const [ isLoggingChecked, setIsLoggingChecked ] = useState( false );
+	const [ customFontValue, setCustomFontValue ] = useState( '' );
+	const [ cssStylingValue, setCssStylingValue ] = useState( '' );
+
+	useEffect( () => {
+		if ( ! isSectionExpanded ) return;
+		if ( ! firstHeadingElementRef.current ) return;
+
+		firstHeadingElementRef.current.focus();
+	}, [ isSectionExpanded ] );
+
+	return (
+		<>
+			<SettingsSection>
+				<Button onClick={ toggleIsSectionExpanded } isTertiary>
+					{ __( 'Advanced settings', 'wordpress-components' ) }
+					<Icon
+						icon={ isSectionExpanded ? chevronUp : chevronDown }
+					/>
+				</Button>
+			</SettingsSection>
+			{ isSectionExpanded && (
+				<SettingsSection>
+					<Card>
+						<CardBody size="large">
+							<h4 ref={ firstHeadingElementRef } tabIndex="-1">
+								Debug mode
+							</h4>
+							<CheckboxControl
+								label={ __(
+									'Log error messages',
+									'woocommerce-payments'
+								) }
+								help={ __(
+									'When enabled, payment error logs will be saved to WooCommerce > Status > Logs.',
+									'woocommerce-payments'
+								) }
+								checked={ isLoggingChecked }
+								onChange={ setIsLoggingChecked }
+							/>
+							<h4>
+								{ __(
+									'Block appearance',
+									'woocommerce-payments'
+								) }
+							</h4>
+							<p>
+								{ __(
+									'Customize or override the appearance of the payment methods block on checkout.',
+									'woocommerce-payments'
+								) }{ ' ' }
+								<ExternalLink href="https://stripe.com/docs/stripe-js/payment-element">
+									{ __(
+										'Learn how',
+										'woocommerce-payments'
+									) }
+								</ExternalLink>
+							</p>
+							<Notice
+								className="advanced-settings__theme-notice"
+								status="warning"
+								isDismissible={ false }
+							>
+								{ __(
+									"We couldn't find your store’s theme font to use in this block " +
+										"so we've replaced it with a fitting alternative. " +
+										"If you want to use your custom font, enter a link to where it's located.",
+									'woocommerce-payments'
+								) }
+							</Notice>
+							<div className="advanced-settings__custom-font-url-wrapper">
+								<CheckboxToggle
+									label={ __(
+										'Use a custom font',
+										'woocommerce-payments'
+									) }
+								>
+									<TextControl
+										label={ __(
+											'Custom font URL',
+											'woocommerce-payments'
+										) }
+										value={ customFontValue }
+										onChange={ setCustomFontValue }
+										autoComplete="off"
+										hideLabelFromVision
+									/>
```

Comment:
What do you think about adding some format check for the font URL here?

Reply:
That's a good idea! I modified the input's `type` to be `url`. I also added a placeholder text to help the user.
I tried adding a `pattern` property, but it doesn't look like the Gutenberg input allows that form of client-side validation.
I added an issue to our backlog to investigate that further 👍